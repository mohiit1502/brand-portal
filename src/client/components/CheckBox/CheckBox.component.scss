
.c-CheckBox {
    .has-error {
       padding-top: 0px !important;
        margin-top: -3px !important;
        color: #dc3545;
    }
}
