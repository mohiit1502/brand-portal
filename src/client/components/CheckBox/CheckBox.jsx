import React from "react";
import PropTypes from "prop-types";
import "./CheckBox.component.scss";
import * as staticContent from "../../images";

const CheckBox = props => {
  return (
    <div className={`c-CheckBox form-row${props.containerClasses ? " " + props.containerClasses : ""}`}>
      <div className={`col${props.colClasses ? " " + props.colClasses : ""}`}>
        <div className="form-check">
          <input type="checkbox" id={props.id} className={`form-check-input${props.checkBoxClasses ? " " + props.checkBoxClasses : ""}`} checked={props.selected}
                 required={props.required} onChange={props.onChange}/>
          <label className={`form-check-label${props.labelClasses ? " " + props.labelClasses : ""}`} htmlFor={props.id}>
            {props.label}
            {props.tou && <span><a href={staticContent.TOU} target="_blank">{props.touLink}</a></span>}
          </label>
        </div>
      </div>
    </div>
  );
};

CheckBox.propTypes = {
  containerClasses: PropTypes.string,
  colClasses: PropTypes.string,
  checkBoxClasses: PropTypes.string,
  id: PropTypes.string,
  labelClasses: PropTypes.string,
  onChange: PropTypes.func,
  parentRef: PropTypes.object,
  required: PropTypes.bool
};

export default CheckBox;
