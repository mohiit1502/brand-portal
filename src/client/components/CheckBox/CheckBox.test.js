import React from 'react';
import CheckBox from './CheckBox';

describe('CheckBox', () => {
   it('renders without error', () => {

   });
});