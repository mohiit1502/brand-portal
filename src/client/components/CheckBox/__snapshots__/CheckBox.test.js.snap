// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders without error 1`] = `
<CheckBox
  checkBoxClasses=""
  colClasses=""
  containerClasses=""
  error=""
  id=""
  label=""
  labelClasses=""
  onChange={[Function]}
  parentRef={Object {}}
  preventHTMLRequiredValidation={true}
  required={true}
  selected={true}
  tou={true}
  touLink=""
>
  <div
    className="c-CheckBox form-row"
  >
    <div
      className="col"
    >
      <div
        className="form-check"
      >
        <input
          checked={true}
          className="form-check-input"
          id=""
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor=""
        >
          <span>
            <a
              href="Walmart Brand Portal Terms of Use 10.25.2021.pdf"
              target="_blank"
            />
          </span>
        </label>
      </div>
    </div>
  </div>
</CheckBox>
`;
